websites/V/Vizzy/presence.ts

```diff
@@ -0,0 +1,129 @@
+const presence = new Presence({
+		clientId: "968353669491871754"
+	}),
+	browsingTimestamp = Math.floor(Date.now());
+function getElementByXpath(path: string) {
+	return document.evaluate(
+		path,
+		document,
+		null,
+		XPathResult.FIRST_ORDERED_NODE_TYPE,
+		null
+	).singleNodeValue;
+}
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			smallImageKey: "logo_512",
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search } = window.location;
+	switch (pathname) {
+		case "/": {
+			presenceData.details = "Viewing the homepage";
+			presenceData.largeImageKey = "home";
+			break;
+		}
+		case "/creations":
+		case "/community": {
+			presenceData.details = "Viewing community creations";
+			presenceData.largeImageKey = "discover";
+			break;
+		}
+		case "/project": {
+			presenceData.largeImageKey = "discover";
+			presenceData.details = "Viewing a project";
+			presenceData.state = getElementByXpath(
+				"/html/body/div[1]/div/main/div/div/div[2]/div/div[2]/div[1]/h4"
+			).textContent;
+			presenceData.buttons = [
+				{
+					label: "View Project",
+					url: location.href
+				}
+			];
+			break;
+		}
+		case "/user": {
+			presenceData.largeImageKey = "discover";
+			presenceData.details = "Viewing a user's page";
+			presenceData.state = `@${
+				getElementByXpath("/html/body/div[1]/div/h5").textContent
+			}`;
+			presenceData.buttons = [
+				{
+					label: "View User",
+					url: location.href
+				}
+			];
+			break;
+		}
+		case "/profile": {
+			presenceData.details = "Viewing their profile";
+			presenceData.largeImageKey = "profile";
+			if (search) {
+				presenceData.state = `Tab: ${new URLSearchParams(
+					document.location.search
```

Comment:
make some sense please
```suggestion
					search
```

Reply:
Cuz it was addressed; replaced with the search variable `tabParams`